### PR TITLE
[BottomNavigation] Use designated initializers.

### PR DESCRIPTION
**Pull request by romoore**
Instead of having only a convenience `-init` method, the BottomNavigationController should implement the UIViewController designated initializers.

Follow-up for #5886
Part of #4160

### CODE REVIEW

**Comment by romoore** [comment]
@jverkoey as reviewer because I'm not sure if we're supporting `initWithCoder:` any longer.  I suspect that the current version will certainly fail because any previously-encoded VC will have duplicated subviews.

**Comment by jverkoey** [comment]
We are supporting initWithCoder: in order to support interface builder. We are not, however, coding properties of our classes.

File: components/BottomNavigation/examples/MDCBottomNavigationBarController.m [hunk 1]
```diff
@@ -38,6 +38,17 @@ - (instancetype)init {
...
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self = [super initWithCoder:aDecoder];
+  if (self) {
+    _navigationBar = [[MDCBottomNavigationBar alloc] init];
+    _content = [[UIView alloc] init];
+    _viewControllers = @[];
+    _selectedIndex = NSNotFound;
+  }
+  return self;
+}
```
Comment (jverkoey) [inline components/BottomNavigation/examples/MDCBottomNavigationBarController.m:51]: Please pull the common init logic to a shared initializer helper method.

**Review by jverkoey — APPROVED** [review]

**Comment by romoore** [comment]
Ok, so if clients end up with duplicated BottomNavigationBar subviews, we don't consider that a supportable bug?

**Comment by jverkoey** [comment]
> Ok, so if clients end up with duplicated BottomNavigationBar subviews, we don't consider that a supportable bug?

Hm where a view is getting duplicated here?  Can you share an example of what you're concerned about?

File: components/BottomNavigation/examples/MDCBottomNavigationBarController.m (hunk 1 above)
Comment (romoore) [reply components/BottomNavigation/examples/MDCBottomNavigationBarController.m:51]: Done

**Comment by romoore** [comment]
The superclass (UIViewController) will decode its `view` property, which will decode its subviews. Those subviews include an MDCBottomNavigationBar object. Once we detect `loadView` being called, we will create a new MDCBottomNavigationBar and add it to the subviews as well.

**Review by nokeeo — APPROVED** [review]

**Comment by romoore** [comment]
Closing this until we can figure out how to get Example swift code to depend on Example ObjC code.

**Pull request closed without merging**